apps/cowswap-frontend/src/legacy/utils/trade.ts

```diff
@@ -116,8 +116,10 @@ export function getSignOrderParams(params: PostOrderParams): SignOrderParams {
     throw new Error(`Order params invalid sellToken address for token: ${JSON.stringify(sellToken, undefined, 2)}`)
   }
 
+  const isSellTrade = kind === OrderKind.SELL
+
   // fee adjusted input amount
-  const sellAmount = (swapZeroFee ? sellAmountBeforeFee : inputAmount).quotient.toString(RADIX_DECIMAL)
+  const sellAmount = (swapZeroFee && isSellTrade ? sellAmountBeforeFee : inputAmount).quotient.toString(RADIX_DECIMAL)
```

Comment:
This is an actual change to buy order behaviour not behind a feature flag.
Are we sure we understand the consequences and it's not breaking anything?

Reply:
Yes, this is an adjustment for the previous fix: https://github.com/cowprotocol/cowswap/pull/3469/files#diff-f4f143c5b037385f28e75ae0ac6e12341e77457e86208671a0127ebe91ed2c1fR114

So, before fee=zero feature we used `inputAmount` as `sellAmount` always.
Now, only  when `swapZeroFee=true` and `isSellTrade=true` we use sellAmountBeforeFee. In fact, the change is under the feature flag